MatrixSDK/MXSession.m

```diff
@@ -2635,9 +2637,9 @@ - (MXHTTPOperation*)uploadDirectRoomsInOperationsQueue:(NSDictionary<NSString *,
                                                failure:(void (^)(NSError *))failure
 {
     // If there is no change, do nothing
-    if (_directRooms == directRooms
-        || [_directRooms isEqualToDictionary:directRooms]
-        || (_directRooms == nil && directRooms.count == 0))
+    if (self.directRooms == directRooms
```

Comment:
Same remark for a temporary variable to host `self.directRooms`

Reply:
👍